tests/bench/runtime_coreapis/main.c

```diff
@@ -58,6 +76,102 @@ static void _flag_waitone(void)
     thread_flags_wait_one(_flag);
 }
 
+static clist_node_t clist;
+
+static void _insert_reserved_data(clist_node_t *list)
+{
+    clist_node_t *last = list->next;
+    if (!last) {
+        return;
+    }
+    unsigned val = BENCH_CLIST_SORT_TEST_NODES;
+
+    clist_node_t *cur = last->next;
+    while (cur != last) {
+        struct test_node *n = container_of(cur, struct test_node, list);
+
+        /* This is the worst case for most sorting algorithms: having the
+         * list perfectly sorted but in inverse direction. For real time, we
+         * only care about worst case and not about best case / average case */
+        n->value = val--;
```

Comment:
This is *not* the worst case for *this* insertion sort implementation, as using `val++` makes sorting *much* slower.

```suggestion
        n->value = val++;
```

E.g., on nrf52840dk, for `n=256`:

before:

```
2025-04-12 23:18:21,904 #  clist_sort (#256, worst):    311423us  ---  311.423us per call  ---       3211 calls per sec
```

after:
```
2025-04-12 23:19:11,954 #  clist_sort (#256, worst):   9233345us  ---  9233.345us per call  ---        108 calls per sec
```

That's a factor 30 slower than the claimed "worst case".

The cut-off point where this insertion sort becomes slower than the previous implementation (which does not come from IOCCC) is below `n=32`.